src/main/java/org/candlepin/subscriptions/tally/SnapshotSummaryProducer.java

```diff
@@ -61,19 +65,32 @@ protected SnapshotSummaryProducer(
   public void produceTallySummaryMessages(Map<String, List<TallySnapshot>> newAndUpdatedSnapshots) {
     AtomicInteger totalTallies = new AtomicInteger();
     newAndUpdatedSnapshots.forEach(
-        (orgId, snapshots) ->
-            snapshots.stream()
-                .sorted(Comparator.comparing(TallySnapshot::getSnapshotDate))
-                .map(snapshot -> summaryMapper.mapSnapshots(orgId, List.of(snapshot)))
-                .forEach(
-                    summary -> {
-                      if (validateTallySummary(summary)) {
-                        kafkaRetryTemplate.execute(
-                            ctx ->
-                                tallySummaryKafkaTemplate.send(tallySummaryTopic, orgId, summary));
-                        totalTallies.getAndIncrement();
-                      }
-                    }));
+        (orgId, snapshots) -> {
+          // Filter snapshots, as we only deal with hourly and non Any fields when we transmit the
+          // tally summary message to the BillableUsage component.
+          List<TallySnapshot> filteredSnapshots =
+              snapshots.stream()
+                  .filter(
+                      snapshot ->
+                          snapshot.getGranularity().equals(Granularity.HOURLY)
+                              && !snapshot.getServiceLevel().equals(ServiceLevel._ANY)
+                              && !snapshot.getUsage().equals(Usage._ANY)
+                              && !snapshot.getBillingProvider().equals(BillingProvider._ANY)
+                              && !snapshot.getBillingAccountId().equals("_ANY"))
```

Comment:
nitpick: use ResourceUtils.ANY instead

Reply:
Yup didn't realize we have this class :D will change it.